### ISSUE TITLE
[Support]: Fatal Python Error: Illegal Instruction

### ISSUE THREAD

**Issue by btoconnor** [blakeblackshear/frigate#1999]
### Describe the problem you are having

I am trying to get a basic frigate setup running on my Ubuntu 20.04 server.  This is a very old AMD Phenom II processor, and I've had issues with it not supporting instructions in the past.  I am hoping to just launch frigate to play around with it before investing in the Coral USB accessory.

Every time I try and launch frigate via docker-compose, I get errors, namely `Fatal Python Error: Illegal Instruction`.  Full output is below.

Log output included below

I am using the docker image `blakeblackshear/frigate:stable-amd64` in my `docker-compose.yml` file.

### Version

unknown?

### Frigate config file

```yaml
mqtt:
  host: mqtt.lan

cameras:
  wyze:
    ffmpeg:
      inputs:
        - path: rtsp://user:pass@<wyzeipaddr>/live
          roles:
            - detect
            - rtmp
      #hwaccel_args:
      #  - /dev/dri/renderD128
    detect:
      width: 1920
      height: 1080

cpu1:
    type: cpu
    num_threads: 3
```

### Relevant log output

```shell
frigate    | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
frigate    | [s6-init] ensuring user provided files have correct perms...exited 0.
frigate    | [fix-attrs.d] applying ownership & permissions fixes...
frigate    | [fix-attrs.d] done.
frigate    | [cont-init.d] executing container initialization scripts...
frigate    | [cont-init.d] done.
frigate    | [services.d] starting services
frigate    | [services.d] done.
frigate    | Fatal Python error: Illegal instruction
frigate    |
frigate    | Current thread 0x00007fbcf43c3740 (most recent call first):
frigate    |   File "<frozen importlib._bootstrap>", line 219 in _call_with_frames_removed
frigate    |   File "<frozen importlib._bootstrap_external>", line 1166 in create_module
frigate    |   File "<frozen importlib._bootstrap>", line 556 in module_from_spec
frigate    |   File "<frozen importlib._bootstrap>", line 657 in _load_unlocked
frigate    |   File "<frozen importlib._bootstrap>", line 975 in _find_and_load_unlocked
frigate    |   File "<frozen importlib._bootstrap>", line 991 in _find_and_load
frigate    |   File "<frozen importlib._bootstrap>", line 219 in _call_with_frames_removed
frigate    |   File "<frozen importlib._bootstrap>", line 1042 in _handle_fromlist
frigate    |   File "/usr/lib/python3/dist-packages/tflite_runtime/interpreter.py", line 41 in <module>
frigate    |   File "<frozen importlib._bootstrap>", line 219 in _call_with_frames_removed
frigate    |   File "<frozen importlib._bootstrap_external>", line 848 in exec_module
frigate    |   File "<frozen importlib._bootstrap>", line 671 in _load_unlocked
frigate    |   File "<frozen importlib._bootstrap>", line 975 in _find_and_load_unlocked
frigate    |   File "<frozen importlib._bootstrap>", line 991 in _find_and_load
frigate    |   File "/opt/frigate/frigate/edgetpu.py", line 12 in <module>
frigate    |   File "<frozen importlib._bootstrap>", line 219 in _call_with_frames_removed
frigate    |   File "<frozen importlib._bootstrap_external>", line 848 in exec_module
frigate    |   File "<frozen importlib._bootstrap>", line 671 in _load_unlocked
frigate    |   File "<frozen importlib._bootstrap>", line 975 in _find_and_load_unlocked
frigate    |   File "<frozen importlib._bootstrap>", line 991 in _find_and_load
frigate    |   File "/opt/frigate/frigate/config.py", line 16 in <module>
frigate    |   File "<frozen importlib._bootstrap>", line 219 in _call_with_frames_removed
frigate    |   File "<frozen importlib._bootstrap_external>", line 848 in exec_module
frigate    |   File "<frozen importlib._bootstrap>", line 671 in _load_unlocked
frigate    |   File "<frozen importlib._bootstrap>", line 975 in _find_and_load_unlocked
frigate    |   File "<frozen importlib._bootstrap>", line 991 in _find_and_load
frigate    |   File "/opt/frigate/frigate/app.py", line 16 in <module>
frigate    |   File "<frozen importlib._bootstrap>", line 219 in _call_with_frames_removed
frigate    |   File "<frozen importlib._bootstrap_external>", line 848 in exec_module
frigate    |   File "<frozen importlib._bootstrap>", line 671 in _load_unlocked
frigate    |   File "<frozen importlib._bootstrap>", line 975 in _find_and_load_unlocked
frigate    |   File "<frozen importlib._bootstrap>", line 991 in _find_and_load
frigate    |   File "/opt/frigate/frigate/__main__.py", line 9 in <module>
frigate    |   File "/usr/lib/python3.8/runpy.py", line 87 in _run_code
frigate    |   File "/usr/lib/python3.8/runpy.py", line 194 in _run_module_as_main
frigate    | [cmd] python3 exited 260
frigate    | [cont-finish.d] executing container finish scripts...
frigate    | [cont-finish.d] done.
frigate    | [s6-finish] waiting for services.
frigate    | [s6-finish] sending all processes the TERM signal.
```

### FFprobe output from your camera

```shell
ffprobe version 4.2.4-1ubuntu0.1 Copyright (c) 2007-2020 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.3.0-10ubuntu2)
  configuration: --prefix=/usr --extra-version=1ubuntu0.1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-nvenc --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
Input #0, rtsp, from 'rtsp://<redacted>/live':
  Metadata:
    title           : Session streamed by the WYZE Media Server
    comment         : live
  Duration: N/A, start: 0.000250, bitrate: N/A
    Stream #0:0: Video: h264 (Main), yuv420p(progressive), 1920x1080, 15 fps, 15.42 tbr, 90k tbn, 30 tbc
    Stream #0:1: Audio: pcm_alaw, 8000 Hz, 1 channels, s16, 64 kb/s
```

### Operating system

Other Linux

### Install method

Docker Compose

### Coral version

CPU (no coral)

### Network connection

Wired

### Camera make and model

Wyze V2 running RTSP firmware

**Comment by 95Kevid** [comment]
> In case anyone stumbles on this, I was able to create a docker image for AMD64 CPUs that don't support the AVX instruction set. I've been running it for several weeks now and it's been mostly very reliable. It is heavily based on [pdecat's Dockerfile in their hass-addons repo](https://github.com/pdecat/frigate-hass-addons/blob/main/frigate_oldcpu/Dockerfile).
> 
> I'm happy to share my Dockerfile if anyone would like it, but it's very similar to the one linked above. I am having some difficulties with `release-grouper` on my CPU and am relying on `release-frogfish` at the moment.
> 
> Obviously, in an ideal world, I wouldn't be running this on a CPU this old - however, with the chip / equipment shortages we're dealing with, I think it makes sense to eek out as much life in these older, still capable machines.

Any chance you could share this?

**Comment by btoconnor** [comment]
Sure - this is the Dockerfile I'm using: https://gist.github.com/btoconnor/e7bbd0c1f6727c235fa6db0c6377c4b4

I don't work with Docker that often, so if there's something I've messed up, feel free to correct me.  This has been working with me for several months now though.

There's a couple of comments in there if you want to try and get grouper running instead of frogfish. I don't know what the practical implications are of using frogfish vs grouper, I just know the last time I tried to get grouper working, it was causing problems on my specific machine.
